### PR TITLE
Fix spurious warning in TF TokenClassification models

### PR DESCRIPTION
**Pull request by Rocketknight1**
This small fix switches `warnings.warn` for `tf.print` in the loss function for TF `TokenClassification` models. When TF compiles a model which has a conditional that depends on values inside a Tensor, it traces both branches. The result of this is that when one of the branches contains a `warnings.warn` statement, that warning is fired during tracing, before the model has seen any data.

Using `tf.print` instead of `warnings.warn` has a few downsides - in particular, it will possibly fire more than once. Still, this is better than firing the warning every time the model is used even if the data is correct, which is what happens now.

### CODE REVIEW

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def hf_compute_loss(self, labels, logits):
         # make sure only labels that are not equal to -100
         # are taken into account as loss
         if tf.math.reduce_any(labels == -1):
-            warnings.warn("Using `-1` to mask the loss for the token is deprecated. Please use `-100` instead.")
+            tf.print("Using `-1` to mask the loss for the token is deprecated. Please use `-100` instead.")
```
Comment (sgugger) [inline src/transformers/modeling_tf_utils.py:222]: Why not use our `logger` with `logger.warn`?

Comment (Rocketknight1) [reply src/transformers/modeling_tf_utils.py:222]: Ah, I didn't realize our logger was a TF logger that wrote graph ops! Let me try that.

Comment (Rocketknight1) [reply src/transformers/modeling_tf_utils.py:222]: Update: Turns out our TF logger doesn't emit graph ops, so the warnings start firing again when I try that:
image

Comment (sgugger) [reply src/transformers/modeling_tf_utils.py:222]: Ok to use `tf.print` then.

Comment (Rocketknight1) [reply src/transformers/modeling_tf_utils.py:222]: Cool!

**Review by gante — COMMENTED** [review]
👍 

For future reference: I was checking with @Rocketknight1 how TF handles deprecation warnings, and they keep a global variable to only print once ([example](https://github.com/tensorflow/tensorflow/blob/87462bfac761435a46641ff2f10ad0b6e5414a4b/tensorflow/python/util/deprecation.py#L220))

**Review by gante — APPROVED** [review]

**Comment by sgugger** [comment]
I think if we use `FutureWarning` they are only printed once, no? Could you check in this instance @Rocketknight1 ?

**Comment by Rocketknight1** [comment]
@sgugger Either `FutureWarning` or `warnings.warn` are correctly only printed once - the problem is that they are **always** printed once, even if there are no -1s in the labels! This is because TF compiles the branch of the graph where the labels tensor contains `-1` and fires the warning during tracing. Only a TF graph op (like `tf.print()`) is correctly fired when the tensor contains `-1`.

**Comment by sgugger** [comment]
Ah thanks for explaining! I had completely misunderstood :-)

**Comment by Rocketknight1** [comment]
Don't worry, even the TF engineers don't understand TF
